sphinx_thebe/__init__.py

```diff
@@ -75,41 +61,10 @@ def update_thebe_context(app, doctree, docname):
         else:
             kernel_name = "python3"
 
-    # Codemirror syntax
-    cm_language = kernel_name
-    if "python" in cm_language:
-        cm_language = "python"
-    elif cm_language == "ir":
-        cm_language = "r"
```

Comment:
Codemirror language is in principle kernel-dependent, however I would like to remove this bit. I think that thebe should configure the codemirror language by talking to the kernel instead. @choldgraf do you agree?

Also currently https://github.com/executablebooks/thebe/issues/241 seems to render this irrelevant.

Reply:
Well the main challenge here is that kernels can have all kinds of wacky names (eg, conda-python-3) , while codemirror only uses "canonical" names for languages (eg, "python") so this logic will need to.be somewhere

Reply:
Once thebe is talking with the kernel and it's running, thebe can read from `codemirror_mode` kernel metadata key, without any config. This would avoid flaky heuristics, and would therefore be a more reliable approach. Or am I missing something?

Reply:
that would require thebe to wait for the kernel to be ready and then update the DOM once that finishes, right?

Reply:
You're right, this is poor UX. I would still prefer to make the logic more reliable. In principle, notebook metadata should already have the codemirror mode specified directly. Any idea how we could get it?

An alternative approach is to assume that the necessary kernel is reachable locally in the environment we use to build the docs, and we could inspect that kernel for metadata. Would that be OK?